File: 1492-the-kth-factor-of-n/main.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        
        vector<int> factor_list;
        for (int i = 1; i < n + 1; i++) {
            if ((n % i) == 0) {
                factor_list.push_back(i);
            }
        }

        return (k - 1) >= factor_list.size() ? -1 : factor_list[k - 1];
    }
};

File: 1492-the-kth-factor-of-n/main.rs
impl Solution {
    pub fn kth_factor(n: i32, k: i32) -> i32 {
        
        let mut factor_list: Vec<i32> = Vec::new();
        for idx in 1..(n+1) as usize {
            if n % idx as i32 == 0 {
                factor_list.push(idx as i32);
            }
        }

        if k - 1 >= factor_list.len() as i32 {
            return -1;
        }

        factor_list[k as usize - 1]
    }
}
